presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java

```diff
@@ -797,27 +796,27 @@ public void testCast()
             throws Exception
     {
         for (Boolean value : booleanValues) {
-            assertExecute(generateExpression("cast(%s as boolean)", value), BOOLEAN, value == null ? null : (value ? true : false));
+            assertExecute(generateExpression("cast(%s as boolean)", value), BOOLEAN, (value));
```

Comment:
confirming: most of these changes were opportunistic fixes removing unneeded conditionals, is that right? would be nice to pull that out into a separate commit (can stay in this PR), just to make it easy to isolate the changes that are relevant to json_extract.